components/loading-box/src/index.js

```diff
@@ -0,0 +1,170 @@
+// TODO INSERT A COMMENT REFERENCE TO EXTERNAL URL IF POSSIBLE
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import glamorous from 'glamorous';
+import { CSSTransition } from 'react-transition-group';
+import hexRgb from 'hex-rgb';
+import { Spinner } from '@govuk-react/icons';
+import { GREY_1, WHITE } from 'govuk-colours';
+
+const Wrapper = glamorous.div({
+  position: 'relative',
+});
+
+const Innerwrap = glamorous.div(({
+  timeIn,
+  timeOut,
+  loading,
+  backgroundColor,
+  backgroundColorOpacity,
+}) => ({
+  position: 'absolute',
+  height: '100%',
+  top: 0,
+  right: 0,
+  left: 0,
+  bottom: 0,
+  display: 'flex',
+  justifyContent: 'center',
+  '> svg': {
```

Comment:
I'm having a think about this... thoughts to come

Reply:
This, and `> div` is assuming knowledge of the implementation of the child components. If the child components elements were to change, this would break.

Please read [Component as a selector](https://glamorous.rocks/examples/#component-as-a-selector) example in the Glamorous docs, specifically:

1. how you could use a component selector instead of `> svg` 
2. the point that " This is actually intentional because it's not a pattern you should generally follow (one of the nice parts of css-in-js is that it allows you to not worry about the cascade of CSS)"

Further to (2), is there any way we could achieve the same without styling an external component? E.g. creating a Glamorous component internal to this file and styling that with props?

Reply:
I would prefer not having a cascading style, but could live with a component as a selector 😀